fontc/src/main.rs

```diff
@@ -265,31 +311,71 @@ fn main() -> Result<(), Error> {
     let prev_inputs = init(&config).map_err(Error::IoError)?;
 
     let mut change_detector = ChangeDetector::new(config.clone(), paths.clone(), prev_inputs)?;
-    let context = Context::new_root(
-        config.args.emit_ir,
-        paths,
-        change_detector.current_inputs.clone(),
-    );
-
-    // TODO: consider making a more explicit model of a graph
-    // TODO: Rayon focuses on CPU-bound tasks but we are doing IO too
-    // perhaps we should do the IO first, e.g. read (but not parse) inputs, etc?
-
-    // TODO: enter metadata work onto threadpool, proper dependencies
-    // Pending proper graph processing just run it first
-    let mut work: Vec<Box<dyn Work>> = Vec::new();
-    add_static_metadata_work(&mut change_detector, &context, &mut work)?;
-    do_work(&context, work)?;
-
-    // Accumulate work for the parts of IR that trivially parallelize
-    let mut work: Vec<Box<dyn Work>> = Vec::new();
+    let mut pending_jobs = create_jobs(&mut change_detector)?;
+    let job_count = pending_jobs.len();
+
+    // Async work will send us it's ID on completion
+    let (send, recv) = crossbeam_channel::unbounded::<(WorkIdentifier, Result<(), WorkError>)>();
```

Comment:
I feel like for something like this I would probably choose a design where there is some sort of `WorkContext` struct or similar that has a `run()` method, and then the runloop logic would live there? Not a blocker, but something you might consider.